corehq/apps/geospatial/views.py

```diff
@@ -446,13 +448,71 @@ class CasesReassignmentView(BaseDomainView):
     def post(self, request, domain, *args, **kwargs):
         request_data = json.loads(request.body)
         case_id_to_owner_id = request_data.get('case_id_to_owner_id', {})
+        include_related_case = request_data.get('include_related_case')
         if len(case_id_to_owner_id) > self.MAX_REASSIGNMENT_REQUEST_CASES:
             return HttpResponseBadRequest(
                 _("Maximum number of cases that can be reassigned is {limit}").format(
-                    self.MAX_REASSIGNMENT_REQUEST_CASES
+                    limit=self.MAX_REASSIGNMENT_REQUEST_CASES
                 )
             )
 
+        if include_related_case:
+            request_cases_id = list(case_id_to_owner_id.keys())
+            parent_to_child_cases_id = self.get_child_cases(domain, request_cases_id)
+            for parent_case_id, child_cases_id in parent_to_child_cases_id.items():
+                for child_case_id in child_cases_id:
+                    self._add_related_case(case_id_to_owner_id, parent_case_id, child_case_id)
+
+            child_to_parent_case_id = self.get_parent_cases(domain, request_cases_id)
+            for child_case_id, parent_case_id in child_to_parent_case_id.items():
+                self._add_related_case(case_id_to_owner_id, child_case_id, parent_case_id)
+
         update_cases_owner(domain, case_id_to_owner_id)
 
         return JsonResponse({'status': 'success'})
+
+    @staticmethod
```

Comment:
Thanks for the explanation. These functions are all being called from the instance itself so I don't believe it would be creating instance copies but rather simply pass itself as `self` in this context.

> Do you think there are any downsides for the static method here ?

This is a tricky one. While I don't believe there are any technical downsides for using a static method here, in terms of readability it can give the impression that this is a function that can be used outside of the instance itself (which conflicts with denoting it as private with _). But I do feel this is getting very nitpicky, and overall I'm okay with leaving this as is. I was just curious to hear your reasoning for making them static.

Reply:
Thanks for your thoughts.  Very helpful.
>  so I don't believe it would be creating instance copies

Ah I meant to say each instance will have their own copy of method vs a single copy for static method.

> in terms of readability it can give the impression that this is a function that can be used outside of the instance itself (which conflicts with denoting it as private with _).

This is a good point. 
I agree on this and will make changes to have instance methods.

Reply:
Addressed in 77dea3b0fdeafe8ecae2aeb912bd64e3d5b7d82d

Reply:
Thanks for thinking through this Ajeet. Changing them to instance methods sounds good.